scripts/utils/schemas.py

```diff
@@ -0,0 +1,144 @@
+
+params_validator = And(
+    Use(parse_value),
+    TrackParamsFile(PARAMS_FILE),
+    error=f'"parameters_file" to be assigned to "{PARAMS_FILE}"',
+)
+additional_files_validator = And(
+    Use(parse_value),
+    TrackAdditionalFiles(Regex(rf"^{ADD_PATH}{os.path.sep}")),
+    error=f'Additional inputs must point to files in "{ADD_PATH}"',
+)
+
+extra_inputs = {
+    Optional("parameters_file"): params_validator,
+    Optional(str): additional_files_validator,
+}
+
+# Define the four supported mlcube schemas
+
+wild_card = {Optional(str): object}
+
+PREP_MLCUBE_TASKS = Schema(
+    {
+        "prepare": {
+            "parameters": {
+                "inputs": {"data_path": object, "labels_path": object, **extra_inputs},
+                "outputs": {"output_path": object},
+            },
+            **wild_card,
+        },
+        "sanity_check": {
+            "parameters": {
+                "inputs": {"data_path": object, **extra_inputs},
+            },
+            **wild_card,
+        },
+        "statistics": {
+            "parameters": {
+                "inputs": {"data_path": object, **extra_inputs},
+                "outputs": {"output_path": object},
+            },
+            **wild_card,
+        },
+        **wild_card,
+    },
+    name="Data Preparation MLCube",
+)
+
+SEPARATE_LABELS_PREP_MLCUBE_TASKS = Schema(
+    {
+        "prepare": {
+            "parameters": {
+                "inputs": {"data_path": object, "labels_path": object, **extra_inputs},
+                "outputs": {"output_path": object, "output_labels_path": object},
+            },
+            **wild_card,
+        },
+        "sanity_check": {
+            "parameters": {
+                "inputs": {"data_path": object, "labels_path": object, **extra_inputs}
+            },
+            **wild_card,
+        },
+        "statistics": {
+            "parameters": {
+                "inputs": {"data_path": object, "labels_path": object, **extra_inputs},
+                "outputs": {"output_path": object},
+            },
+            **wild_card,
+        },
+        **wild_card,
+    },
+    name="Separate Labels Data Preparation MLCube",
+)
+
+
+MODEL_MLCUBE_TASKS = Schema(
+    {
+        "infer": {
+            "parameters": {
+                "inputs": {"data_path": object, **extra_inputs},
+                "outputs": {"output_path": object},
+            },
+            **wild_card,
+        },
+        **wild_card,
+    },
+    name="Model MLCube",
+)
+
+METRICS_MLCUBE_TASKS = Schema(
+    {
+        "evaluate": {
+            "parameters": {
+                "inputs": {"predictions": object, "labels": object, **extra_inputs},
+                "outputs": {"output_path": object},
+            },
+            **wild_card,
+        },
+        **wild_card,
+    },
+    name="Metrics MLCube",
+)
+
+# Put the available schemas in a dict
+schemas = {
+    "prep": PREP_MLCUBE_TASKS,
```

Comment:
If these are the mlcube types that are exposed to the user, I'd probably use the complete names for prep `data-preparator` and `data-preparator-separate-labels`. Might be a mouthful, but I think it's best to be clear and explicit here